aten/src/ATen/TensorImpl.cpp

```diff
@@ -55,62 +55,67 @@ void Tensor::backward(
   pImpl->backward(std::move(gradient), keep_graph, create_graph);
 }
 
-TensorImpl::TensorImpl(Backend backend, ScalarType scalar_type) {
-  backend_ = backend;
-  scalar_type_ = scalar_type;
-  auto type = &globalContext().getType(backend, scalar_type);
-  Storage* storage = type->storage(true).release();
-  StorageImpl* storage_impl = storage->pImpl();
-  tensor = new THTensor(storage_impl);
-}
-
-TensorImpl::~TensorImpl() {
-  if (tensor) tensor->release();
+TensorImpl::TensorImpl(
+    Backend backend,
+    ScalarType scalar_type,
+    bool is_variable)
+    : TensorImpl(nullptr, backend, scalar_type, is_variable) {
+  auto type = &globalContext().getType(backend_, scalar_type_);
+  try {
```

Comment:
This should be resolved! Not clear what the best approach is:
- Could add virtual "make_storage" to Type and have subclasses return nullptr
- Could add argument "has_storage"